src/Endpoints/class-metadata-endpoint.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Metadata endpoint abstract class
+ *
+ * @package Parsely\Endpoints
+ * @since 3.2.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Endpoints;
+
+use Parsely\Parsely;
+
+/**
+ * Metadata endpoint classes are expected to implement the remaining functions of the class.
+ *
+ * @since 3.2.0
+ */
+abstract class Metadata_Endpoint {
+	protected const VERSION    = '1.0.0';
+	protected const FIELD_NAME = 'parsely';
+
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	protected $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Function to start up the class and enqueue necessary actions.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return void
+	 */
+	abstract public function run(): void;
+
+	/**
+	 * Registers the metadata fields on the appropriate resource types.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @return void
+	 */
+	abstract public function register_meta(): void;
+
+	/**
+	 * Get the metadata in string format.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @return string String containing the metadata as HTML code that can be directly inserted in into the page.
+	 */
+	public function get_rendered_meta(): string {
+		ob_start();
+		$this->parsely->insert_page_header_metadata();
```

Comment:
Is this still suitable generic for other endpoints? What if there's data in a REST API endpoint that is something we don't make available in the page header data, like Settings values etc.

I think this `get_rendered_meta()` method should probably stay in the Rest_Metadata class for now.

Reply:
Thinking about the future `GraphQL_Metadata` class, this is for sure a method we'll reuse. I have some experiments going on in https://github.com/Parsely/wp-parsely/pull/706 for some additional context.